### DISCUSSION TITLE
how can i close zk client curator EnsembleTracker in dubbo

### DISCUSSION THREAD

**Discussion by xull1995** [apache/dubbo#14232] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

Background：
our dubbo client and zk deploy in k8s，but not at the same k8s。 the zk cluster use Headless service as zk server address，expose loaderbalancer ip for dubbo client use。

When the Dubbo client reconnects to Zookeeper, the reconnection fails because it cannot resolve the headless domain name.

The reasons is：

1、curator in 3.0.0 version  add EnsembleTracker, when zk client start ,will get /zookeeper/config znode data ,and change user defined zk IP address,  at this time ,the ip changed to headless domain。

so ,how can i close the curator curator EnsembleTracker in dubbo ?  

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
Is curator support so?

**Comment by xull1995** [comment]
yes , please see  : https://github.com/apache/curator/blob/master/curator-framework/src/main/java/org/apache/curator/framework/CuratorFrameworkFactory.java#L278

**Reply by AlbumenJ** [reply]
Would you please submit a PR to enhance it?

You can update the configuration in `org.apache.dubbo.remoting.zookeeper.curator5.Curator5ZookeeperClient#Curator5ZookeeperClient`.

**Comment by aofall** [comment]
It has been merge by #14281 

While the new version releases. you can disable EnsembleTracker by config.

```
dubbo:
  application:
    parameters:
      zookeeper.ensemble.tracker: false
```

**Reply by xull1995** [reply]
thanks!
